apps/hpc-ftsadmin/src/app/utils/parse-flow-form.ts

```diff
@@ -0,0 +1,1119 @@
+    inactiveReason.push(cancelledCategory);
+  }
+
+  const flowObjects = getFundingValues(values).flatMap((key) =>
+    extractDirectionObject(key, values)
+  );
+
+  const flow: flows.CreateFlowParams['flow'] = {
+    activeStatus: !isInactive || !!isPending?.isApproved,
+    amountUSD: currencyToInteger(amountUSD),
+    budgetYear:
+      donorBudgetYear !== '' ? valueToInteger(donorBudgetYear) : undefined,
+    categories,
+    children: childFlows.map((childFlow) => ({ childID: childFlow.id })),
+    decisionDate: decisionDate?.toDate() ?? null,
+    description,
+    exchangeRate: exchangeRate ? valueToFloat(exchangeRate) : null,
+    firstReportedDate: firstReported.toDate(),
+    flowDate: flowDate.toDate(),
+    flowObjects,
+    isCancellation: null, //  TODO
+    isErrorCorrection:
+      isErrorCorrection || !!isPending?.isApproved || isPending?.isSaved,
+    isApprovedFlowVersion: !!isPending?.isApproved || isPending?.isSaved,
+    inactiveReason,
+    newCategories: [], //  TODO
+    newMoney: isNewMoney,
+    notes,
+    origAmount: amountOriginalCurrency
+      ? currencyToInteger(amountOriginalCurrency)
+      : null,
+    origCurrency: currency?.value.toString() ?? null,
+    parents: parentFlow ? [{ parentID: parentFlow.id }] : [],
+    reportDetails: reportingDetailPropsToReportDetails(
+      reportingDetails,
+      initialValues
+    ),
+    restricted: isRestricted,
+  };
+  return { flow };
+};
+
+const isCategoryGroupKeyFlowForm = (
+  group: string
+): group is keyof FlowFormType => {
+  return [
+    'flowStatus',
+    'flowType',
+    'earmarkingType',
+    'contributionType',
+    'method',
+    'beneficiaryGroup',
+  ].includes(group);
+};
+
+const categoriesToFlowForm = (values: flows.GetFlowResult) => {
+  return values.categories.reduce((acc, { group, id, name }) => {
+    if (group === 'keywords') {
+      return {
+        ...acc,
+        [group]: [...(acc.keywords ?? []), { displayLabel: name, value: id }],
+      };
+    } else if (isCategoryGroupKeyFlowForm(group)) {
+      const parsedValue: util.FormObjectValue = {
+        displayLabel: name,
+        value: id,
+      };
+      if (group === 'method' && !isMethodOption(parsedValue)) {
+        return { ...acc, childMethod: parsedValue };
+      }
+      return { ...acc, [group]: parsedValue };
+    }
+    return acc;
+  }, {} as FlowFormType);
+};
+
+const isInferred = (
+  flow: flows.GetFlowResult,
+  entity: { id: number },
+  direction: RefDirection,
+  entityName: EntityName
+) => {
+  if (flow.externalReferences.length === 0) {
+    return false;
+  }
+
+  const inferredList = flow.externalReferences.map((eR) => {
+    if (!eR.importInformation?.inferred) {
+      return false;
+    }
+    return eR.importInformation.inferred.some(
+      (inf) =>
+        inf.key.includes(direction) &&
+        inf.key.includes(entityName) &&
+        inf.valueId === entity.id
+    );
+  });
+  return inferredList.includes(true);
```

Comment:
This is more efficient, since it exits the first time `true` is found
```suggestion
  return flow.externalReferences.some((eR) => {
    if (!eR.importInformation?.inferred) {
      return false;
    }
    return eR.importInformation.inferred.some(
      (inf) =>
        inf.key.includes(direction) &&
        inf.key.includes(entityName) &&
        inf.valueId === entity.id
    );
  });
```